source/common/dmtbdump2.c

```diff
@@ -1681,39 +1681,43 @@ AcpiDmDumpPhat (
     ACPI_PHAT_HEADER        *Subtable;
     ACPI_PHAT_VERSION_DATA  *VersionData;
     UINT32                  RecordCount;
-    UINT32                  Length = Table->Length;
+    UINT32                  TableLength = AcpiUtReadUint32 (&Table->Length);
     UINT32                  Offset = sizeof (ACPI_TABLE_PHAT);
     UINT32                  SubtableLength;
     UINT32                  PathLength;
     UINT32                  VendorLength;
+    UINT16                  SubtableType;
 
 
     Subtable = ACPI_ADD_PTR (ACPI_PHAT_HEADER, Table, sizeof (ACPI_TABLE_PHAT));
 
-    while (Offset < Table->Length)
+    while (Offset < TableLength)
     {
         /* Common subtable header */
 
+        SubtableType = AcpiUtReadUint16 (&Subtable->Type);
+	Subtable->Type = SubtableType;
```

Comment:
This line causes a segfault with 'iasl -T phat' and needs to be removed.  GCC 11 is the compiler, on Fedora33, and I suspect a compiler problem at this point.  Regardless, this is extraneous code and is not needed so removing the line has no effect.